beacon-chain/core/blocks/block_operations.go

```diff
@@ -84,3 +89,183 @@ func verifyProposerSlashing(
+//     vote.aggregate_signature_poc_1_indices.
+//   Let intersection = [x for x in indices(casper_slashing.votes_1)
+//     if x in indices(casper_slashing.votes_2)].
+//   Verify that len(intersection) >= 1.
+//   Verify that casper_slashing.votes_1.data.justified_slot + 1 <
+//     casper_slashing.votes_2.data.justified_slot + 1 ==
+//     casper_slashing.votes_2.data.slot < casper_slashing.votes_1.data.slot
+//     or casper_slashing.votes_1.data.slot == casper_slashing.votes_2.data.slot.
+//   For each validator index i in intersection,
+//     if state.validator_registry[i].status does not equal
+//     EXITED_WITH_PENALTY, then run
+//     update_validator_status(state, i, new_status=EXITED_WITH_PENALTY)
+func ProcessCasperSlashings(
+	validatorRegistry []*pb.ValidatorRecord,
+	casperSlashings []*pb.CasperSlashing,
+	currentSlot uint64,
+) ([]*pb.ValidatorRecord, error) {
+	if uint64(len(casperSlashings)) > params.BeaconConfig().MaxCasperSlashings {
+		return nil, fmt.Errorf(
+			"number of casper slashings (%d) exceeds allowed threshold of %d",
+			len(casperSlashings),
+			params.BeaconConfig().MaxCasperSlashings,
+		)
+	}
+	for idx, slashing := range casperSlashings {
+		// verifyCasperSlashing performs validity checks on the slashing
+		// struct and returns a list of validator indices to be slashed from
+		// the BeaconState's validator registry.
+		validatorIndices, err := verifyCasperSlashing(slashing)
+		if err != nil {
+			return nil, fmt.Errorf("could not verify casper slashing #%d: %v", idx, err)
+		}
+		for _, validatorIndex := range validatorIndices {
+			penalizedValidator := validatorRegistry[validatorIndex]
+			if penalizedValidator.Status != pb.ValidatorRecord_EXITED_WITH_PENALTY {
+				// TODO(#781): Replace with
+				// update_validator_status(
+				//   state,
+				//   validatorIndex,
+				//   new_status=EXITED_WITH_PENALTY,
+				// ) after update_validator_status is implemented.
+				validatorRegistry[validatorIndex] = v.ExitValidator(
+					penalizedValidator,
+					currentSlot,
+					true, /* penalize */
+				)
+			}
+		}
+	}
+	return validatorRegistry, nil
+}
+
+func verifyCasperSlashing(
+	slashing *pb.CasperSlashing,
+) ([]uint32, error) {
+	votes1 := slashing.GetVotes_1()
+	votes2 := slashing.GetVotes_2()
+	votes1Indices := append(
+		votes1.GetAggregateSignaturePoc_0Indices(),
+		votes1.GetAggregateSignaturePoc_1Indices()...,
+	)
+	votes2Indices := append(
+		votes2.GetAggregateSignaturePoc_0Indices(),
+		votes2.GetAggregateSignaturePoc_1Indices()...,
+	)
+
+	votes1Attestation := votes1.GetData()
+	votes2Attestation := votes2.GetData()
+
+	if err := verifyCasperVotes(votes1); err != nil {
+		return nil, fmt.Errorf("could not verify casper votes 1: %v", err)
+	}
+	if err := verifyCasperVotes(votes2); err != nil {
+		return nil, fmt.Errorf("could not verify casper votes 2: %v", err)
+	}
+
+	// Inner attestation data structures for the votes should not be equal.
```

Comment:
Why not? If you could add context here, that would be great. I don't immediately see the answer in the spec. Maybe Danny knows.